### ISSUE TITLE
Add option to close Datagrid filter by clicking outside of its popup.

### ISSUE THREAD

**Issue by sdurnov** [vmware/clarity#2705]
## Summary

Here is a closed request to make filters close when clicking outside of a filter popup. https://github.com/vmware/clarity/issues/2679

* **What is the change?** Instead of changing current behaviour, I'd ask to add an option on a Datagrid level to control such behavior.
* **Why should it go in Clarity?** Because current design isn't confortable and desirable for all users. It's not very comfortable to be limited with "x" button in each filter. It requres extra click when you're setting multiple filters, and an argument that user wants to see all filters he set isn't working for everybody. For instance, we're displaying current filter above the grid in a card. It lets us to keep user aware of current filter if it was pre-selected. I think this is a pretty viable solution, at least in comparison with the current design.
* **Does this change impact existing behaviors? If so how?** It won't. Having this mode as a non-default option will keep already existing code and workflows untouched, but will let people to have a choice.

## Use case

Filters are being summarized outside of a Datagrid in a card, or text string, or with a Label components.
This is very useful in case user setting multiple filters for a complex grid, or especially, if grid has a feature of filter presets.

## Workarounds

If there is a feature to turn such behaviour on, in current API, or, there is a way to implement that with extra bit of code, it would be OK.

**Comment by yurenwang** [comment]
I totally agree with this idea. Currently it is seriously uncomfortable to click on the tiny "x" every time in order to close a filter; and opening and closing filters are such commonly used.
